x/concentrated-liquidity/fees_test.go

```diff
@@ -699,3 +701,125 @@ func (s *KeeperTestSuite) TestCollectFees() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestUpdateFeeAccumulatorPosition() {
+	var (
+		ownerOne = s.TestAccs[0]
+		ownerTwo = s.TestAccs[1]
+	)
+
+	type updateFeeAccumPositionTest struct {
+		poolId        uint64
+		owner         sdk.AccAddress
+		liquidity     sdk.Dec
+		lowerTick     int64
+		upperTick     int64
+		expectedError error
+	}
+
+	positions := map[int]updateFeeAccumPositionTest{
+		1: {
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick,
+		},
+		2: {
+			lowerTick: DefaultLowerTick + 1,
+			upperTick: DefaultUpperTick,
+		},
+		3: {
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick + 1,
+		},
+	}
+
+	tests := map[string]updateFeeAccumPositionTest{
+		"update position with a different pool ID": {
+			poolId:    2,
+			owner:     ownerOne,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick,
+		},
+		"update position with a different owner": {
+			poolId:    1,
+			owner:     ownerTwo,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick,
+		},
+		"update position with a different lower tick": {
+			poolId:    1,
+			owner:     ownerOne,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick + 1,
+			upperTick: DefaultUpperTick,
+		},
+		"update position with a different upper tick": {
+			poolId:    1,
+			owner:     ownerOne,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick + 1,
+		},
+		"err: pool does not exist": {
+			poolId:        3,
+			expectedError: cltypes.PoolNotFoundError{PoolId: 3},
+		},
+		"err: position does not exist": {
+			poolId:        1,
+			owner:         ownerOne,
+			liquidity:     DefaultLiquidityAmt,
+			lowerTick:     DefaultLowerTick - 1,
+			upperTick:     DefaultUpperTick,
+			expectedError: accum.NoPositionError{Name: fmt.Sprintf("%s,%s,%d,%d", "1", ownerOne.String(), DefaultLowerTick-1, DefaultUpperTick)},
+		},
+	}
```

Comment:
wdyt of adding test cases for trying to updateFeeAccumulatorPosition for pool that does not fee accumulator?

Reply:
So we currently test for positions that dont have a fee accumulator set. I am not sure if it should be the job of this function to ensure fee accumulator is set at the pool level. Unless I am misunderstanding, that would require us to make a new function to create a cl pool specifically so we dont initialize a fee accumulator.

Reply:
This makes sense to me!